pkg/registry/seedmanagement/managedseed/strategy.go

```diff
@@ -65,12 +67,31 @@ func (s Strategy) PrepareForUpdate(ctx context.Context, obj, old runtime.Object)
 	oldManagedSeed := old.(*seedmanagement.ManagedSeed)
 	newManagedSeed.Status = oldManagedSeed.Status
 
-	if !apiequality.Semantic.DeepEqual(oldManagedSeed.Spec, newManagedSeed.Spec) ||
-		oldManagedSeed.DeletionTimestamp == nil && newManagedSeed.DeletionTimestamp != nil {
+	if mustIncreaseGeneration(oldManagedSeed, newManagedSeed) {
 		newManagedSeed.Generation = oldManagedSeed.Generation + 1
 	}
 }
 
+func mustIncreaseGeneration(oldManagedSeed, newManagedSeed *seedmanagement.ManagedSeed) bool {
+	// The spec changed
+	if !apiequality.Semantic.DeepEqual(oldManagedSeed.Spec, newManagedSeed.Spec) {
+		return true
+	}
+
+	// The deletion timestamp was set
+	if oldManagedSeed.DeletionTimestamp == nil && newManagedSeed.DeletionTimestamp != nil {
+		return true
+	}
+
+	// The operation annotation was added with value "reconcile"
```

Comment:
Discussion regarding style:
For me comments are always a burden as they are in danger to be outdated when the code below changes but the comment was forgotten to be changed. In this case I think the function names already provide enough meaning so the comments are unnecessary. WDYT?
(this comment should not hold merging the PR back)

Reply:
> comments are always a burden as they are in danger to be outdated when the code below changes but the comment was forgotten to be changed

I don't do this. If there is a comment, I update it also (perhaps I do forget sometimes, but very rarely). If there is not, I often add one. I am a "comment addict" since in more than 20 years of practice I've found comments help me much more often than mislead me. I believe the same is true for most people and the danger of comments becoming outdated and misleading is greatly overrated.

This particular piece of code is quite similar to other `mustIncreaseGeneration` functions, e.g. https://github.com/gardener/gardener/blob/ff684eab504da137bd4b8ed5670f6108f3c400d9/pkg/registry/core/shoot/strategy.go#L71-L112. They are also full of comments. So also for the sake of consistency with the rest of the code, I think the comments should be kept.